# <span style="color: #2455C3;">Binomial distribution</span>

In [49]:
import math
from scipy.stats import binom

$$P\left ( x \text{ out of } n \right ) = B\left ( x, n, p \right ) = \left ( \begin{matrix}
n\\ 
x
\end{matrix} \right ) p^x (1-p)^{n-x}$$

### <span style="color: #2455C3;">Basic example</span>

You say "70% choose chicken, so 7 of the next 10 customers shoudl choose chicken" ... what are the chances you are right?

* p = 0.7
* n = 10
* x = 7

So we get:

$$p^x (1-p)^{n-x} = 0.7^7 (1-0.7)^{10-7} = 0.77(0.3)(3) = 0.0022235661$$

that is the probability of each outcome. So the total probability that you're right is:

$$P\left ( 7 \text{ out of } 10 \right ) = B\left ( 7, 10, 0.7 \right ) = \left ( \begin{matrix}
10\\ 
7
\end{matrix} \right ) 0.7^7 (1-0.7)^{10-7} = 0.266827932$$

In fact the **probability of 7 out of the next 10** choosing chicken in only about **27%**

**Moral of the story**: even though the long-run average is 70% don't expect 7 out of the next 10.

In [11]:
p = 0.7
n = 10
x = 7

binom.pmf(x, n, p)

0.26682793200000016

## <span style="color: #2455C3;">Practice examples</span>

### <span style="color: #2455C3;">1. Throw the die</span>

A fair die is thrown four times. Calculate the probabilities of getting 2 twos.

In [22]:
p = 1/6.0 # because the die has six posible outcomes and it is fair
n = 4
x = 2

prob = binom.pmf(x, n, p)
print "{:0.2f}%".format(prob*100)

11.57%


**for the 4 throws there are 11.57% chances of 2 twos**

### <span style="color: #2455C3;">2. Sports bike</span>

Your company makes sports bikes, 90% pass final inspection (and 10% fail and need to be fixed). What is the Mean and Variance of the 4 next inspections?

In [23]:
n = 4
p = 0.9

print "Mean: {:0.2f}".format(n*p)
print "Variance: {:0.2f}".format(n*p*(1-p))

Mean: 3.60
Variance: 0.36


In [34]:
print "x    P(x)    x+P(x)"
mean = 0
for x in range(n+1):
    prob = binom.pmf(x, n, p)
    mean += x*prob
    print "{}    {:4.4f}  {:0.4f}".format(x, prob, x*prob)
print
print "     SUM    {:0.4f}".format(mean)

x    P(x)    x+P(x)
0    0.0001  0.0000
1    0.0036  0.0036
2    0.0486  0.0972
3    0.2916  0.8748
4    0.6561  2.6244

     SUM    3.6000


### <span style="color: #2455C3;">3. Fair coin</span>

A fair coin is tossed five times. What is the probability of obtaining two heads?

In [43]:
p = 0.5
n = 5
x = 2

prob = binom.pmf(x, n, p)
print "{:0.2f}%".format(prob*100)

31.25%


### <span style="color: #2455C3;">4. Fair coin 2 </span>

A fair coin is tossed 16 times. What is the mean and standard deviation number of Heads?

In [50]:
p = 0.5
n = 16

mean = n*p
variance = n*p*(1-p)

print "Mean: {:0.2f}".format(mean)
print "Variance: {:0.2f}".format(variance)
print "Variance: {:0.2f}".format(math.sqrt(variance))

Mean: 8.00
Variance: 4.00
Variance: 2.00


### <span style="color: #2455C3;">5. Electronic components production</span>

A company makes electronic components for TV's. 95% pass final inspection (and 5% fail and need to be fixed). 120 components are inspected in one day. What is the expected number that fail in one day?

In [51]:
p = 0.05
n = 120

mean = n*p
variance = n*p*(1-p)

print "Mean: {:0.2f}".format(mean)
print "Variance: {:0.2f}".format(variance)
print "Variance: {:0.2f}".format(math.sqrt(variance))

Mean: 6.00
Variance: 5.70
Variance: 2.39


### <span style="color: #2455C3;">6. Fair coin 3</span>

A fair coin is tossed five times. Use the binomial probability formula to calculate the probability of **at least four heads**.

In [71]:
p = 0.5
n = 5

prob = sum(binom.pmf(x, n, p) for x in range(4, n+1))

print "{:0.2f}%".format(prob*100)

18.75%


### <span style="color: #2455C3;">7. Fair coin 4</span>

A fair coin is tossed six times. Use the binomial probability formula to calculate the probability of **at most two tails**.

In [73]:
p = 0.5
n = 6

prob = sum(binom.pmf(x, n, p) for x in range(3))

print "{:0.2f}%".format(prob*100)

34.38%


### <span style="color: #2455C3;">8. Fair die</span>

A fair cubical die is thrown four times. Use the binomial probability formula to calculate the probability of **at least two 3's**.

In [78]:
p = 1/6.0
n = 4

prob = sum(binom.pmf(x, n, p) for x in range(2, n+1))

print "{:0.2f}%".format(prob*100)

13.19%
